reactioncommerce/reaction
# Implement `canSeeAdminUI` check

Signed-off-by: trojanh <rajan.tiwari@kiprosh.com>

Resolves https://github.com/reactioncommerce/reaction-admin/issues/192 
Impact: **breaking**  
Type: **feature**

## Issue
Currently the admin UI always shows but for non-admins, data will be missing and some things will be broken due to API authorization checks.

## Solution
Add a global setting canSeeAdminUI in the API, and in Reaction Admin:

Add a way to toggle it (boolean) from settings
Hide the UI when it's false. Show a nice message, perhaps linking to the storefront.

## Breaking changes
none


## Testing
1. Open GraphQL playground
2. Check if `canSeeAdminUI` field is visible in `GlobalSettings` query and `updateGlobalSettings` mutation.
